Issue #4 🎫: MusicAlbum class

ITurres opened the pull request:
# 🚩I have made the following changes to complete Issue #4 

## Here is a summary of what has been done

### Type of **`files`** changed/added

![My Skills](https://skillicons.dev/icons?i=ruby)

### In this new Branch, I have **➕ADDED➕** the succeeding **\*📂 folders** and/or **\*\*📄 files** to complete this ticket's [requirements](https://github.com/ITurres/catalog_of_things/issues/4)

# \*📂 root/

- ## \*📂 lib/

  - **\*\*📄 music_album.rb** ➕

    - I have added all the attributes and methods required.

### I have **✏️MODIFIED✏️** the succeeding **\*📂 folders** and/or **📄\*\* files**

# \*📂 root/

- ## \*📂 lib/

   ❗ _**Shared-file**_❗
  - **\*\*📄 item.rb** ✏️

    - I have moved `genre`, `author` and `label` attributes to have accessor.
    This will ensure that when an object of the same name is passed to the `Item` class's `add_[method-name]` it
    properly sets that object as the attribute.

---

### Thank you for taking the time to review this PR ⭐

---

If you need additional information or have any questions, please don't hesitate to contact me on Slack as _**Arturo (Arthur) Emanuel Guerra Iturres**_. I'll be happy to help you out. 🎯

---

ITurres committed:
Fix: Give setter to Item's genre, author, label

- Now when we create a new item, we can set
  its genre, author and label.

ITurres committed:
Add 'music_album.rb / MusicAlbum' class

demesameneshoa reviewed (approved):
Hi @ITurres , 

Your ticket 🎫  is complete!  There is nothing else to say other than it's time to merge it :shipit: 
Congratulations! 🎉 

<img src="https://media.giphy.com/media/8yZuEb7q6dlyP3L3cL/giphy.gif"/>

## Highlights
✅ The `music_album` class is created
✅ The Class is initialized properly
✅ The genre class takes instance of `Item` class
✅ No linters errors
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._



Cheers and Happy coding!👏👏👏

## Have any doubt ❓

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

<table  width="100%" align="center">
  <tr>
    <td colspan="4" align="center">
      <strong>Contact me: Demes Ameneshoa</strong>
    </td>
  </tr>
  <tr>
    <td align="center">
      <a href="https://github.com/demesameneshoa">
        <img src="https://img.shields.io/badge/github-%23121011.svg?style=for-the-badge&logo=github&logoColor=white" alt="GitHub">
      </a>
    </td>
    <td align="center">
      <a href="https://www.linkedin.com/in/demesameneshoa/">
        <img src="https://img.shields.io/badge/linkedin-%230077B5.svg?style=for-the-badge&logo=linkedin&logoColor=white" alt="LinkedIn">
      </a>
    </td>
    <td align="center">
      <a href="https://microverse-students.slack.com/team/U051695H9RS">
        <img src="https://img.shields.io/badge/Slack-4A154B?style=for-the-badge&logo=slack&logoColor=white" alt="Slack">
      </a>
    </td>
    <td align="center">
      <a href="mailto:demesameneshoa@gmail.com?subject=Microverse%20Code%20Review">
        <img src="https://img.shields.io/badge/Gmail-D14836?style=for-the-badge&logo=gmail&logoColor=white" alt="Email">
      </a>
    </td>
  </tr>
</table>

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

ITurres committed:
Merge branch 'development' into feature/musicAlbum-class

- This merge is due to the `game.rb` PR approved by @demesameneshoa
and merged into `development` branch.

ITurres committed:
Feat: Enhance 'MusicAlbum' class constructor to:
- Accept:
    - A item object with the instantiation of: Genre, Author and Label.
    - super publish_date with default value of nil.

ITurres committed:
Fix: 'Game' class's super 'publish_date' attr typo

ITurres commented:
@demesameneshoa! Heads up! 😄 , I have given the correct number and type of arguments to the `MusicAlbum` constructor as well as a default value to `publish_date` super attr. 

I have also fixed a typo on the super attr of the `Game` class. (This shouldn't have been done in this PR but the team is already aware). Hope it's not an issue!. Let me know.

Cheers! 🥂
